### DISCUSSION TITLE
Selecting multiple slices? Group actions?

### DISCUSSION THREAD

**Discussion by eekboom** [pmndrs/zustand#2016] [Q&A] [answered]
The recipes still contain this, which gives a deprecation warning:
```
const { nuts, honey } = useStore(
  (state) => ({ nuts: state.nuts, honey: state.honey }),
  shallow
)
```
If I understand it correctly, without the shallow, the component would just always refresh, because the state selection function always creates a new object, right?

Most of the time, I only use one or two actual data objects from state, but may use a couple of actions.

So would you think it a good approach, to
* fall back to atomic state picks for data
* group all actions under an action key, so that I can pick multiple actions like
```
const {addVote, removeVote, updateAnswer} = useStore(state => state.actions)
```
Because my actions property never changes, that should never cause a component rerender, right?

**Comment by dai-shi** [comment, marked as answer]
Sounds good. Only a caveat would be we need custom merge for persist middleware.

**Reply by eekboom** [reply]
Thanks a lot for the quick answer!
Currently, I am doing axios requests in my actions "manually". Seems to work fine and I guess I just don't have that problem then.
Will have a deeper look at the persistence middleware. Let me learn one thing at a time ;-)
Also I'd like to keep complexity and dependencies low.
